### ISSUE TITLE
[META] Xcode 14.3 Support for Xamarin, .NET 6/7 and MAUI

### ISSUE THREAD

**Issue by dalexsoto** [xamarin/xamarin-macios#17561]
🚨🚨 Update 5 🚨🚨

Support for Xcode 14.3 has now shipped as part of [Visual Studio 2022 version 17.6 release](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#visual-studio-2022-version-176), [Visual Studio for Mac 2022 version 17.5.5](https://learn.microsoft.com/en-us/visualstudio/releases/2022/mac-release-notes#1755--visual-studio-2022-version-1755) or using the links below.

|  Product  | Version | Package                                | Changelog |
|-----------|---------|----------------------------------------|-----------|
|Xamarin.iOS|16.4.0.6| [xamarin.ios-16.4.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/a90bad45b475c278658a0598d3f5a579/xamarin.ios-16.4.0.6.pkg) |https://github.com/xamarin/xamarin-macios/compare/xamarin-ios-16.2.0.6...xamarin-ios-16.4.0.6|
|Xamarin.Mac| 9.3.0.6| [xamarin.mac-9.3.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/b572dce8bcab6b76cba92e494f56f1e2/xamarin.mac-9.3.0.6.pkg) |  https://github.com/xamarin/xamarin-macios/compare/xamarin-mac-9.1.0.6...xamarin-mac-9.3.0.6|

Thank you for your patience and testing!

The macios team!

<details>
  <summary>Update 4</summary>

🚨🚨 Update 4 🚨🚨

Apple released Xcode 14.3 on March 30th, 2023, we have what we believe are the final builds for you to test on Xamarin.

* [Xamarin.iOS 16.4.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/a90bad45b475c278658a0598d3f5a579/xamarin.ios-16.4.0.6.pkg)
* [Xamarin.Mac 9.3.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/b572dce8bcab6b76cba92e494f56f1e2/xamarin.mac-9.3.0.6.pkg)

For MAUI we are [still producing builds](https://github.com/dotnet/maui/pull/14334) compatible with Xcode 14.3 so we recommend using Xcode 14.2 for now. If you updated to Xcode 14.3 and want to go back to Xcode 14.2 you can download Xcode 14.2 from the [downloads](https://developer.apple.com/download/all/) page in the Apple Developer Portal and continue to use it.

If you use Xcode 14.3 with the current tooling you will get an `mlaunch` error:

> error HE0004: Could not load the framework 'ContentDeliveryServices' (path: /Applications/Xcode.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/ContentDeliveryServices):
dlopen(/Applications/Xcode.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/ContentDeliveryServices, 0x0001): Library not loaded: @rpath/ContentDelivery.framework/Versions/A/ContentDelivery
Referenced from: <77B7BF95-A673-3DD7-875F-6393D0092351> /Applications/Xcode.app/Contents/SharedFrameworks/ContentDeliveryServices.framework/Versions/A/Frameworks/ITunesConnectFoundation.framework/Versions/A/ITunesConnectFoundation

**The solution is to use Xcode 14.2.**

We will keep this issue updated as we go to have the final builds declared.

Thank you for your patience!

The macios team!

</details>

**Comment by dalexsoto** [comment]
Posting an update here: Xcode 14.3 now requires macOS Ventura as noted, so test your build infra as for us has been quite a challenging update due to more strict system permissions. It boils down to identifying what tools you are using that need either Full Disk Access (Open System Settings, click Privacy & Security, then click Full Disk Access.) or Files and folders permissions (Also inside System Settings, click Privacy & Security).

**Comment by happysoftware-es** [comment]
Thanks for the info and your hard work dalexsoto. So, I have a question... where can I download this package: Xamarin.iOS 16.4.0.1.pkg?
Thanks in advance

**Comment by omatrot** [comment]
How to we downgrade to Xcode 14.2?

**Comment by lubiepomaranczki** [comment]
I've just done it and this should do the trick 😇

1. First, you need to uninstall the current version of Xcode from your Mac. Drag the Xcode application from the Applications folder to the Trash to do this.

2. Now, download the Xcode 14.2 version from the Apple Developer website or from the App Store using this direct link: https://developer.apple.com/download/more/?=xcode%2014.2

Once the download is complete, you can open the downloaded file and follow the instructions to install Xcode 14.2.

3. Verify that the Xcode 14.2 version is installed and selected by running the following command in the Terminal app:
`xcodebuild -version`
This command should display the version number of Xcode 14.2.

**Comment by omatrot** [comment]
That worked thanks.
But...
I now have problems to compile my project.
VS still looks for newer SDK:
```
clang : warning : no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.4.sdk' [-Wmissing-sysroot]
clang : warning : no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.4.sdk' [-Wmissing-sysroot]
clang : warning : no such sysroot directory: '/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS16.4.sdk' [-Wmissing-sysroot]
```

**Comment by omatrot** [comment]
Neither of those tricks solved the problem :(
I can see Xcode 14.2 in preferences.

**Comment by omatrot** [comment]
I solved my problem by removing subdirectories from:

```
~/Library/Preferences/VisualStudio/
~/Library/Caches/VisualStudio/
~/Library/Application\ Support/VisualStudio/
```
As suggested in a response to this [question](https://developercommunity.visualstudio.com/t/reset-to-default-settings-in-vs-for-mac/805486).

**Comment by NigelWD** [comment]
#17980

**Comment by NigelWD** [comment]
I downgraded to xcode 14.2 and tried all methods in this thread and am still receiving errors #17980

**Comment by sauravchaudhary94** [comment]
Any update like by when Xcode 14.3 will be compatible with Xamarin(Visual studio)?

**Comment by rolfbjarne** [comment]
> I downgraded to Xcode 14.2 and tried all methods in this thread. I still get the same error! Please help!

Please share the complete error output you get.

**Comment by rolfbjarne** [comment]
> > > I downgraded to Xcode 14.2 and tried all methods in this thread. I still get the same error! Please help!
> > 
> > 
> > Please share the complete error output you get.
> 
> I get the same error: error HE0004: Could not load the framework 'ContentDeliveryServices'

You should only get this error with Xcode 14.3. Please make absolutely sure that you're using Xcode 14.2 - note that we've seen cases where macOS updates any Xcodes on your machine just because, so you might need to actually open up the Xcode you've selected and check the version.

**Comment by Huntes** [comment]
For anyone who is also facing this issue so far: If you have both instances of xcode installed on your MAC (14.3 and 14.2), you can also run your app by changing Locations > Command Line Tools option to XCode 14.2

(In my case I need the latest version 14.3 for another project, but I also need the old 14.2 due to this problem, so I can keep both versions and work around this problem temporarily)

**Comment by nbevans** [comment]
The failure mode of Xamarin.iOS is silent. The simulator list is empty. Builds work fine. I've just spent 2+ hours trying everything to get my simulator list to populate. But now I find this thread and realise its because XCode upgraded itself to 14.3 and that this build isn't support yet by Xamarin.iOS.

Why is Xamarin.iOS not emitting at least a warning saying "this version of XCode is not supported" so that there is some hint at least about why things may be broken?

**Comment by nbevans** [comment]
Hey presto - installed 14.2, changed the Command Line Tools option to 14.2 and then re-connected the Mac Pairing in Visual Studio and now the simulators list is populated

<img width="851" alt="image" src="https://user-images.githubusercontent.com/542188/231167864-1c8aa3c2-d2b1-40a3-b672-f7391132dcd4.png">

**Comment by dejanbasic** [comment]
It works for me with this build [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg) and Xcode 14.3

**Comment by MaxFmi** [comment]
The mentioned version [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg) works with xcode 14.3
Thanks

**Comment by Christophe-Chantraine-djm-digital** [comment]
@dejanbasic , 

It's working perfectly for me with [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg)

**Comment by dejanbasic** [comment]
> > @dalexsoto, @dejanbasic ,
> > 
> > It's working perfectly for me with [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg)
> 
> how are we supposed to just update that one package? I updated VSS to 17.5.4, but still simulators are empty

Just install the package

**Comment by samirgcofficial** [comment]
> > > @dalexsoto, @dejanbasic ,
> > > It's working perfectly for me with [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg)
> > 
> > 
> > how are we supposed to just update that one package? I updated VSS to 17.5.4, but still simulators are empty
> 
> Just click on this link: [Xamarin.iOS 16.4.0.3.pkg](https://download.visualstudio.microsoft.com/download/pr/60490eec-956e-47b3-b063-79cadfe6ef11/b9c977eb4bc99a72f1deed54a680bd67/xamarin.ios-16.4.0.3.pkg), double-click on it and install this package :)

This works :)

**Comment by eXeShnik** [comment]
Debugging works fine, what about launchscreen?
Currently, I have two Xamarin Forms projects. I see a black screen instead of launch one

**Comment by dalexsoto** [comment]
Hello Everyone!

We are glad that the current Xamarin.iOS/Mac packages are working for you all, **thanks a lot❣️** for taking the build for a spin 🙏. You can consider the [above build](https://github.com/xamarin/xamarin-macios/issues/17561#issue-1588218177) as production ready as is the one we intend to ship inside the Visual Studio update. As I mentioned on another post reconciling all the different release schedules between Apple Xcode releases, Visual Studio (Windows and Mac) releases and making sure that other teams on top of us are also working (for example the [MAUI SDK](https://github.com/dotnet/maui/pull/14334)) and of course bring this updated support across the .NET 6, 7 and 8 versions is quite a feat as we want to deliver to you a quality product so we all are working to get you a solid Xcode 14.3 support release.

Thanks a lot for your patience while we go through the entire process.

The macios team!

**Comment by schlaman** [comment]
**Does anyone know if Visual Studio 2022 v17.6.0 Preview 3 for Windows Arm** contains the fix mentioned above:

_**"You can consider the https://github.com/xamarin/xamarin-macios/issues/17561#issue-1588218177 as production ready as is the one we intend to ship inside the Visual Studio update"**_

I am running Visual Studio 2022 v17.6.0 Preview 3 for Windows Arm in Parallels on a Mac M2 Max.
I had to downgrade XCode to 14.2 from 14.3 and I had to downgrade my iPad from 16.4 to 16.3.1 to make everything work.

I see reference to **[Xamarin.iOS 16.4.0.3.pkg]** and **not sure how to install it or what it is** but I understand it will solve the problem until the Visual Studio release that resolves this problem.

Can someone add to this please?

Thanks,
Craig

**Comment by chartierpw** [comment]
All,

I had run into a variety of issues with the simulator and managed, after several days, to correct them all.  First I ran into the issue where none of the simulators were appearing in the VS 2022 dropdown list, so I made sure everything was up to date on my Mac and VS (windows).  All was up to date or at lease that's what  was reported, and I still wasn't seeing the simulators in the VS drop down.  I posted in the VS community forums and was directed here telling me my issue was a duplicate to this, and so it was.  The first thing I did was download and installed the Xamarin.iOS v16.4.0.3 package as suggested by @ dejanbasic @MaxFmi. this got the simulator list back but the simulator was so slow, it was unusable.  I discovered that Hyper-V was not enable on my windows machine, so I enabled it.  Just for giggles, I also check to see that the Virtual Technology was enabled in the BIOS (HP Omen Laptop), it was, but I toggled it to be sure there were no issues.  I then re-started VS 2022, opened my iOS project, cleaned and re-built the project.  Now I was getting a Fatal Error when running the simulator.  I posted THAT on the VS Community forums and was once again directed here as it was also considered a duplicate issue.  I noticed that  @dalexsoto posted a newer version of Xamarin.iOS and Xamarin.Mac.

[Xamarin.iOS 16.4.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/a90bad45b475c278658a0598d3f5a579/xamarin.ios-16.4.0.6.pkg)
    [Xamarin.Mac 9.3.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/b572dce8bcab6b76cba92e494f56f1e2/xamarin.mac-9.3.0.6.pkg)

So I downloaded and installed them on my Mac (Macbook Pro running Ventura).  Did not seem to fix the problem as the simulator was crashing with the same Fatal Error.  So I took a look into the logs to see if there was anything in there that could point me in the right direction and I discovered this: "An attempt was made to access a socket in a way forbidden by its access permissions".  A quick goggle told me that there may be an issue with the "Host Network Service", following the instructions contained in this [post](https://stackoverflow.com/questions/10461257/an-attempt-was-made-to-access-a-socket-in-a-way-forbidden-by-its-access-permissi) I was able to get the simulator to run and execute my iOS project.  It's still a little klugey, especially when entering text into a text field or selecting a option from a spinner, but it executes.  Hopefully this will help anyone having the same issues.

R/
Prescott ....

**Comment by omatrot** [comment]
Can we have the release notes for this version of Xamarin.iOS?

**Comment by robertmaxted** [comment]
When will the fix become an official release? I am currently using Xcode 14.2 however the devices in Xcode (iOS devices) are not compatible with iOS 16.4.1 (on iPhone). this means I can only debug/deploy via cable which is absolutely nuking the battery in the MacBook Pro m1..

**Comment by ninaada** [comment]
Any timeline on the fix?

**Comment by CyberZeus** [comment]
> When will the fix become an official release? I am currently using Xcode 14.2 however the devices in Xcode (iOS devices) are not compatible with iOS 16.4.1 (on iPhone). this means I can only debug/deploy via cable which is absolutely nuking the battery in the MacBook Pro m1..

Xcode (any version( is not currently compatible with iOS 16.4.1! It is compatible with versions up to 16.4.
The fix is installing Xamarin.iOS 16.4.0.6.pkg to be able to debug with a real device.

**Comment by robertmaxted** [comment]
I see, but my iOS device is 16.4.1, will installing Xamarin.iOS 16.4.0.6.pkg solve the issue?..

**Comment by robertmaxted** [comment]
> > When will the fix become an official release? I am currently using Xcode 14.2 however the devices in Xcode (iOS devices) are not compatible with iOS 16.4.1 (on iPhone). this means I can only debug/deploy via cable which is absolutely nuking the battery in the MacBook Pro m1..
> 
> Xcode (any version) is not currently compatible with iOS 16.4.1! It is compatible with versions up to 16.4. The fix is installing Xamarin.iOS 16.4.0.6.pkg to be able to debug with a real device.

You cannot use Xcode 14.2 with the iOS 16.4.0.6.pkg.. So if you have installed the older Xcode 14.2, update to 14.3 before installing the package. Haven't tested yet as 14.3 is currently downloading (all 7.3Gb of it!). Thanks again for the heads up. If it works correctly, I'll update on here.

**Comment by robertmaxted** [comment]
Update: With Xcode 14.3 and 16.4.0.6.pkg installed the debugging on device works, but only works when using cable (which is no different to downgrading to 14.2). Debug over wifi does not work.

**Comment by snelsn04** [comment]
Is there a timeline on a fix?

**Comment by sisaacks** [comment]
I can confirm that installing the Xamarin.iOS 16.4.0.6.pkg worked for me and resolved both issues I was having after upgrading. I always dread upgrading, it seems something always breaks.

**Comment by birdcone** [comment]
Still got generic simulator after install new pkg.  However, per @omatrot's post, I removed cache files via "rm ~/Library/Caches/VisualStudio" which solved the problem and I can now see iOS simulators

**Comment by gdang** [comment]
I installed both `.pkg` files on the MacBook at my workplace and I have the iOS simulators show up again for Xamarin and MAUI projects (VS2022 v17.5.4, Windows).

However, my co-worker still cannot see the simulators for MAUI projects. The simulators do show up for Xamarin projects, though. She's also using VS2022 17.5.4 (Windows).

**Comment by jasells** [comment]
> Update: With Xcode 14.3 and 16.4.0.6.pkg installed the debugging on device works, but only works when using cable (which is no different to downgrading to 14.2). Debug over wifi does not work.

I am in same boat.  I even did a re-install of MacOS back to 13.2, and manual install of XCode 14.2.... still no joy for network debugging on device.  I can debug a sim running on my MacBook from Windows 10/VS 2022 via network.

**Comment by bobwhitten** [comment]
With VS for Win 17.5.4, I loaded XCode 14.3 in my Mac, then after reeading this, Xamarin.iOS 16.4.0.6.pkg &
Xamarin.Mac 9.3.0.6.pkg.  Kept getting "incompatible" mssage on Pair to Mac (offering to reload the old Xamarin.iOS).  Very frustrated til I saw a post elsewhere with the hint:
"click NO and continue"
Now, works perfect, Thought I should share, if this helps someone else.
Looking forward to perm fix.

**Comment by CyberZeus** [comment]
> Update: With Xcode 14.3 and 16.4.0.6.pkg installed the debugging on device works, but only works when using cable (which is no different to downgrading to 14.2). Debug over wifi does not work.

Did you enable debug via network on Device & Simulators of Xcode?

**Comment by robertmaxted** [comment]
> > Update: With Xcode 14.3 and 16.4.0.6.pkg installed the debugging on device works, but only works when using cable (which is no different to downgrading to 14.2). Debug over wifi does not work.
> 
> 
> 
> Did you enable debug via network on Device & Simulators of Xcode?
> 
> On my install I can debug via Wi-Fi with no problems

Yes i enabled debug via network in xcode -> devices and simulators -> iPhone -> box is checked.

**Comment by bikrambhandari48** [comment]
Any idea when this will be resolved? I having an issue where my app crashes on iPad (16.4.1) but works on iPhone (16.4.1). I tried all the above instructions. 

It's causing a nightmare for our production app.

Does anybody have any idea?

**Comment by robertmaxted** [comment]
Anyone know when this will be resolved?

**Comment by sisaacks** [comment]
I am having the same issue again.  Even with the mentioned Xamarin.ios version installed.  This is absolutely frustrating.

**Comment by ramtechjoe** [comment]
Is there a plan to deploy this as part of a Visual Studio update?  There have been a couple updates to VS since this was posted but Visual Studio is still at Xamarin.iOS 16.2.0.5.  I generally like to wait till the release with VS but an inadvertent update of my test iPad to 16.4 puts me in a desperate bind for XCode 14.3

**Comment by LeoJHarris** [comment]
All Xcode releases should be deemed unsafe until proven otherwise as we no longer install Xcode updates when immediately  available. To many compatibility issues arises with updates.

**Comment by Meekohi** [comment]
After downloading and installing [Xamarin.iOS 16.4.0.6.pkg](https://download.visualstudio.microsoft.com/download/pr/ca583156-cb11-4858-99b0-5090c412f26d/a90bad45b475c278658a0598d3f5a579/xamarin.ios-16.4.0.6.pkg) I get the following error:
```
/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/iOS/Xamarin.Shared.targets(4,4): Error MSB4064: The "SdkDevPath" parameter is not supported by the "DSymUtil" task loaded from assembly: Xamarin.iOS.Tasks, Version=16.2.0.5, Culture=neutral, PublicKeyToken=84e04ff9cfb79065 from the path: /Library/Frameworks/Xamarin.iOS.framework/Versions/Current/lib/msbuild/iOS/Xamarin.iOS.Tasks.dll. Verify that the parameter exists on the task, the <UsingTask> points to the correct assembly, and it is a settable public instance property. (MSB4064) (Meekohi.App.iOS)
```
It would be great if this issue could be updated with some information about what the underlying problem actually is, and if there is any path toward a solution at this time -- we've been 3 months without support for the current version of XCode and it isn't clear form the updates if there is a plan for moving forward.

**Comment by mattjohnsonpint** [comment]
Our CI builds are failing now, because we do `dotnet workload install maui-ios` so it installs the latest, which then later fails with:

> ILLINK : warning MT0079: The recommended Xcode version for Microsoft.iOS 16.4.7054 is Xcode 14.3 or later. The current Xcode version (found in /Applications/Xcode_14.2.app/Contents/Developer) is 14.2.

This is on a Github Actions macOS runner.  I can't get Xcode 14.3 because GitHub Actions only offers macOS 12 runner images presently.

So what to do?  How do I install the previous version of the `maui-ios` workload (or even just the `ios` workload) that is compatible with Xcode 14.2?

**Comment by mickbyrne** [comment]
Hey @dalexsoto, this ticket was opened nearly 3 months ago now. Any idea when the latest version of Xcode will be supported? We have other fully-native projects that require XCode 14.3 and doing this manual install is a bit of a hassle.

**Comment by LeoJHarris** [comment]
@mickbyrne try using https://github.com/XcodesOrg/XcodesApp its quite a simple application that helps managing multiple version of xcode. Just override your VS instance to use the older version of Xcode and use the latest Xcode for native dev. Should be relatively simple.

**Comment by dpozimski** [comment]
Hi Xamarin Team!
Is there any chance to fix the issue in current and future versions?

**Comment by Phenek** [comment]
Here a feature request for Visual Studio for Mac

[🚨🚨Support informations for Xamarin MAUI](https://developercommunity.visualstudio.com/t/%F0%9F%9A%A8%F0%9F%9A%A8Support-informations-for-Xamarin-MAU/10347765)

**Comment by Phenek** [comment]
I would like to take this opportunity to express some concerns and suggestions regarding the ongoing challenges we face in managing the compatibility between Visual Studio, Xcode, and the Android SDK.

So I open a thread on the Developer community.
[🚨🚨Support informations for Xamarin MAUI](https://developercommunity.visualstudio.com/t/%F0%9F%9A%A8%F0%9F%9A%A8Support-informations-for-Xamarin-MAU/10347765)

Best regards,

**Comment by mattjohnsonpint** [comment]
What's really needed IMHO is that the workloads themselves have an understanding of which Xcode versions they're compatible with, such that `dotnet workload install ios` (et al.) will install the latest that's compatible with the installed Xcode and macOS version, rather than the just the latest available.

**Comment by kyurkchyan** [comment]
I have downgraded to XCode 14.2 + Xamarin.iOS SDK 16.2.0.5 which can be downloaded from [here](https://github.com/xamarin/xamarin-macios/blob/main/DOWNLOADS.md) and the issue was resolved. But it's frustrating to realize that I had to spend at least 2 hours on this... I wish there were at least more explicit error messages suggesting the incompatibility and direct download links to resolve the issue.

**Comment by TPitman-mindfiretechnology** [comment]
It looks like Microsoft just released a new version of iOS workloads to support XCode 14.3. This works to build and deploy, but now I get this error:

`error MT1007: Failed to launch the application ... Invalid Service Error (error: 0xe8000022)
        . You can still launch the application manually by tapping on it.`

**Comment by sisaacks** [comment]
@TPitman-mindfiretechnology I would roll back to the iOS version described above in here.  That was the error I was getting also and was also manually able to launch the app.  Its uber frustrating.  Like @kyurkchyan I spent probably 2-3 hours trying to figure out what was going on.  I dread updating anything when it comes to iOS

**Comment by bobwhitten** [comment]
@sisaacks, I understand why people say wait.  But, with Apple's iOS updates, it often pushes you to a new XCode.  Plus, backing up to previous version of XCode is inconvenient, and previous iOS is super-inconvenient.  But, It's nice to have the latest for use (and to make sure nothing went bad).
AGREE THAT CLEAR IDENTIFICATION OF WHAT XCODE VERSION GOES WITH WHAT WOULD BE VERY HELPFUL!

**Comment by TPitman-mindfiretechnology** [comment]
One weird thing is that on an older iPhone with version 15 it starts fine with the same latest xcode and ios workloads.

It is only on my iPhone 13 with the latest 16 that requires xcode 14.3 that it has a problem starting the app.

**Comment by ac2112** [comment]
> It looks like Microsoft just released a new version of iOS workloads to support XCode 14.3. This works to build and deploy, but now I get this error:
> 
> `error MT1007: Failed to launch the application ... Invalid Service Error (error: 0xe8000022) . You can still launch the application manually by tapping on it.`
> 
> I am able to manually launch it and it seems to run and allow debugging, so it is inconvenient, but not a show stopper.

This appeared right after Apple released a security update. The only way to debug your app is to manually start the app. I've had debugging issues developing this way that didn't exist before.

**Comment by LeoJHarris** [comment]
@ac2112 also seeing the same problem with debugging detaching when app is launched.

**Comment by lilunxm12** [comment]
VS 17.5 bundled Xamarin.iOS 16.4.0.6 that seems only works with Xcode 14.3, which requires MacOS 13 and renders older deivices unusable.

**Comment by scriptBoris** [comment]
> VS 17.5 bundled Xamarin.iOS 16.4.0.6 that seems only works with Xcode 14.3, which requires MacOS 13 and renders older deivices unusable.

Also faced this problem. I have an old mac air 2015, it won't let me install xcode14.3, which makes development for ios impossible.
It is necessary to write documentation for solving this problem

**Comment by Phenek** [comment]
@scriptBoris
On this point the Apple documentation is very clear [Xcode support - here](https://developer.apple.com/support/xcode/) 

I would suggest you  [OpenCore legacy patcher](https://dortania.github.io/OpenCore-Legacy-Patcher/) to install the last Mac OS on old Mac.
But it is out of topic.

**Comment by lilunxm12** [comment]
@Phenek 
Nobody said Apple documentation is unclear. Just want to point out that this may be an unplanned/communicated deprecation? I didn't pay much attention to this repo,  just from the release notes, I get that Xcode 13.4 support is added but no where mentioned previous Xcode versions (and by extension OS and hardware) support were removed

**Comment by ramtechjoe** [comment]
You would like that this post might be updated to indicate the newest VS release includes this version of Xamarin.iOS.  Also would have been nice if release notes for the VS release indicate it supports XCode 14.3.

**Comment by scriptBoris** [comment]
> @lilunxm12 @scriptBoris On this point the Apple documentation is very clear [Xcode support - here](https://developer.apple.com/support/xcode/)
> 
> I would suggest you [OpenCore legacy patcher](https://dortania.github.io/OpenCore-Legacy-Patcher/) to install the last Mac OS on old Mac. But it is out of topic.

OpenCare patcher is helpful me, thank you big man @lilunxm12!

**Comment by scriptBoris** [comment]
> I figured out a workaround to pin the previous version:
> 
> Create `rollback.json` file:
> 
> ```json
> {
>   "microsoft.net.sdk.ios": "16.2.2054/7.0.100",
>   "microsoft.net.sdk.maccatalyst": "16.2.2054/7.0.100"
> }
> ```
> 
> Specify the rollback file when installing workloads. Example:
> 
> ```
> dotnet workload install maui-ios maui-maccatalyst --from-rollback-file rollback.json
> ```
> 
> This presumes using the .NET 7 SDK. My `global.json` is:
> 
> ```json
> {
>   "sdk": {
>     "version": "7.0.100",
>     "rollForward": "latestMinor",
>     "allowPrerelease": false
>   }
> }
> ```
> 
> [`16.2.2054`](https://www.nuget.org/packages/Microsoft.NET.Sdk.iOS.Manifest-7.0.100/16.2.2054) appears to be the latest version published that supports Xcode 14.2 on macOS 12.

That solution is not worked for me

**Comment by beeradmoore** [comment]
Looks like 16.4.0.6 just hit VS4Mac stable branch

**Comment by TPitman-mindfiretechnology** [comment]
Sadly it has the issues with being able to write to an iOS device. When you deploy, it doesn't overwrite the ipa on the device and then doesn't start debugging because the app on the phone is out of date.

If you delete the app from the device before deploying, then it will deploy the app, but then on some phones still won't start the app and you have to start it manually.

**Comment by velocitysystems** [comment]
Anyone else having the problem that @TPitman-mindfiretechnology has reported?

**Comment by SBurgos00** [comment]
I'm using Xamarin.Forms 5.0.0.2578. I just updated to VS 2022 17.6 on Windows and Xcode 14.3. VS updated the mac. 

I did a debug and deploy of my project (on an iPad, iOS 16.4.1 (a)) and it will build, deploy and start the app, even over an existing app. it looks like the debugger is connected but it doesn't break on anything, and after a couple minutes the debugger disconnects with the app still running ("The app has been terminated.). If I try to stop debugging before it automatically disconnects, VS hangs. After a minute or so it gives an error ("Debugger was unable to terminate one or more processes") but the app is still running. This is all a remote debug from a device connected to a mac.

I had to restart VS for it to see a locally connected iPad. I uninstalled the app. I ran debug, it built and deployed the app. VS asked me to start the app manually. I did but it didn't start debugging - the popup message didn't go away until it "disconnected" like it did earlier connected to the mac ("The app has been terminated.").

So for me it can build, deploy and start from the mac but there is no debugging period. This is different from what @TPitman-mindfiretechnology experienced. They mentioned a phone but I'm using an iPad. I don't know what else could be different between our setups.

**Comment by SBurgos00** [comment]
I updated Xamarin.Forms to 5.0.0.2599-pre1. if I try to debug on the iPad connected to my PC, the build fails:
The "UnpackFrameworks" task failed unexpectedly.
System.IO.IOException: The file 'obj\iPhone\Debug\device-builds\ipad6.11-16.4.1/Frameworks\MSAL.framework\Info.plist' already exists.		

If I connect the iPad to the mac and debug remote, it build, deploys and debug works. It looks like updating everything and using a mac connected iPad works. I hope it works for others out there.

**Comment by TPitman-mindfiretechnology** [comment]
I think I know what the difference is.

My issues are somehow related to my app be a MAUI app. I would think the same flow would be used, but something must be different.

I also have Xamarin.Forms apps and those are able to download and debug just fine with the latest stuff.

My MAUI app, however, has the issues I have reported.

**Comment by Syed-RI** [comment]
@dalexsoto @davidortinau Any ETA when MAUI folks can deploy and debug to a physical device using Xcode 14.3 and iOS 16.4?

**Comment by TPitman-mindfiretechnology** [comment]
@Syed-RI the latest does allow that, but has issues as I have reported. If you remove your app before deploy, it will download to your device. Depending on the device OS version you may have to start it manually.

It works, but isn't the full experience yet.

**Comment by dalexsoto** [comment]
Support for Xcode 14.3 has now shipped as part of [Visual Studio 2022 version 17.6 release](https://learn.microsoft.com/en-us/visualstudio/releases/2022/release-notes#visual-studio-2022-version-176) and [Visual Studio for Mac 2022 version 17.5.5](https://learn.microsoft.com/en-us/visualstudio/releases/2022/mac-release-notes#1755--visual-studio-2022-version-1755).

This issue will be locked now that Xcode 14.3 support is out on all IDEs. Please if you encounter any issues with the release file a [new bug](https://github.com/xamarin/xamarin-macios/issues/new) and fill in all the details. I see that some of you are experiencing some debugging issues with the IDE, please do file a ticket via the IDE you are using ([Windows](https://developercommunity.visualstudio.com/VisualStudio/report) or [Mac](https://developercommunity.visualstudio.com/VisualStudioMac/report)) so the issue reaches the right team.

Thank you all for your patience and testing!

The macios team!
